epam_tr/src/main/java/com/epam/tr/service/FileSystemService.java

```diff
@@ -20,46 +20,54 @@
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
-import static com.epam.tr.dto.FileDto.FileSystemObjectType.*;
+import static com.epam.tr.dto.FileDto.FileDtoType.*;
 import static java.io.File.listRoots;
 import static java.util.Arrays.stream;
 import static java.util.Objects.requireNonNull;
 
 @Service
 public class FileSystemService implements FileService {
 
-    @Autowired
-    private Validator<FileDto> validator;
     @Autowired
     private FileEntityBuilder builder;
     @Autowired
     private PathBuilder pathBuilder;
 
     @Override
-    public void create(FileRequestDto requestDto) throws IOException {
+    public void create(FileRequestDto requestDto)  {
         FileDto fileDto = builder.buildFileEntity(requestDto);
-        validator.isValid(fileDto);
         if (fileDto.getType() == FOLDER) {
-            createFolder(fileDto);
+            try {
+                createFolder(fileDto);
+            } catch (IOException e) {
+                throw new InvalidFileException(e.getMessage());
```

Comment:
uniformity??
look what you've done for createFolder and createFile, why is it different?
I like createFile option.